pkg/ibmcsidriver/node.go

```diff
@@ -379,12 +376,20 @@ func (csiNS *CSINodeServer) NodeGetInfo(ctx context.Context, req *csi.NodeGetInf
 		},
 	}
 
-	// maxVolumesPerNode is the maximum number of volumes attachable to a node; default is 4
-	cores := runtime.NumCPU()
-	if cores >= MinimumCoresWithMaximumAttachableVolumes {
-		maxVolumesPerNode = MaxVolumesPerNode
+	// If environment variable is set, use this value as maxVolumesPerNode
+	value, ok := os.LookupEnv("VOLUME_ATTACHMENT_LIMIT")
+	if !ok {
+		ctxLogger.Warn("Invalid value for VOLUME_ATTACHMENT_LIMIT. Setting the default value:", zap.Reflect("MaxVolumesPerNode", maxVolumesPerNode))
+	} else {
+		volumeAttachmentLimit, err := strconv.ParseInt(value, 10, 64)
+		if err != nil {
+			ctxLogger.Warn("Invalid value for VOLUME_ATTACHMENT_LIMIT. Setting the default value:", zap.Reflect("MaxVolumesPerNode", maxVolumesPerNode))
+		} else {
+			maxVolumesPerNode = volumeAttachmentLimit
+		}
 	}
-	ctxLogger.Info("Number of cores of the node and attachable volume limits.", zap.Reflect("Cores", cores), zap.Reflect("AttachableVolumeLimits", maxVolumesPerNode))
+
+	ctxLogger.Info("Number of cores of the node and attachable volume limits.", zap.Reflect("AttachableVolumeLimits", maxVolumesPerNode))
```

Comment:
Please change the log here only to- "Attachable volume limit"

Reply:
done